### ISSUE TITLE
Team Schema

### ISSUE THREAD

**Issue by jeffreyxchan** [ideahacks/ideahacks.la#24]
**Summary**
Last year, we recorded people's names on an excel sheet, and that's how we formed teams during the hackathon. This year, I'm hoping to support a team system natively within our application. This will involve storing information about teams into our database. Therefore we'll need to define what a team is by creating a team schema.

Task: Create the Team schema. Will be another file in the /db folder called Team.js. Export it out of the db module just like you did for Users. Remember branch naming standard.

**Good Resources**
Mongoose Populate documentation: http://mongoosejs.com/docs/populate.html

**Team Schema**
teamName: String  
teamNumber: Number  
parts: [ String ] // this is how you declare an array of strings.
members: [{ type: Schema.Types.ObjectId, ref: 'User' }] // yes this is how you declare it

So we have a String, a Number, and an array of Strings. But you might be asking, what the heck is that last field?

In SQL/Relational database, you often store fields that reference an entirely different table within the database. For example, you can have a table of teams, and each teams has users, but it doesn't have all the user's information, just like their full name or something. Where is the rest of the user information? It'll probably be stored in some huge user table with all possible information about them. SQL/Relational databases can are built to connect tables like a team table and a user table. When working with team information, an algorithm might want to look at a teammate and be like "hey I want more information on this teammate". TLDR SQL/Relational database make it really easy for you to go to the user table and look up information about a user.

MongoDB is a NoSQL non-relational database. Instead of tables, it's built from "documents". This brings up some problems, because maybe when we're working with teams, we'll want to pull up information about a teammate within the team. NoSQL is not built with with the intention of being able to connect information across documents.

But mongoose actually provides us with some tools to add relational properties to our MongoDB. The method is called "populating" (see resource attached above) and essentially lets us use ObjectId's to reference information in other documents. If we ever need to pull information about users, we can take an array of ObjectId's of User objects and "populate" the array so that the array is actually filled with user objects instead. I probably butchered it up a lot, but it's a pretty neat ability to have, and will prevent a bunch of headaches.

Long story short: the "members" field will be an array of ObjectId's that reference User objects.

ObjectId's are built into MongoDB objects. If there exists a user in the User database, it will have an ObjectId by default, because Mongo gives every object an ObjectId.
